test/avr/test_lib/UnixStampTestAvr.cpp

```diff
@@ -0,0 +1,112 @@
+#include <unity.h>
+#include <UnixStamp.hpp>
+#include <stdint.h>
+
+#define EXPECETED_CHERNOBYL_TIMESTAMP 514848225
+#define EXPECTED_ERA_END 951868799
+#define EXPECTED_ERA_START 951868800
+#define EXPECTED_EPOCH_LEFT_BORDER 0
+#define EXPECTED_EPOCH_RIGHT_BORDER 4294967295
+
+void setUp(void) {}
+
+void tearDown(void) {}
+
+civil_time fillTimeStruct(int16_t year, int8_t mon, int8_t day, int8_t hour, int8_t min, int8_t sec, int8_t tz)
+{
+    civil_time time;
+    time.year = year;
+    time.mon = mon;
+    time.day = day;
+    time.hour = hour;
+    time.min = min;
+    time.sec = sec;
+    time.tz = tz;
+    return time;
+}
```

Comment:
Check out [Aggregate initialization](https://en.cppreference.com/w/cpp/language/aggregate_initialization). This function doesn't offer much on top of the basic aggregate initialization (`T object = { arg1, arg2, ... };`).

And, if C++20 features are available with your toolchain, an even better alternatives are

```cpp
T object = { .des1 = arg1 , .des2 { arg2 } ... }; 	// (3) 	(since C++20)
T object { .des1 = arg1 , .des2 { arg2 } ... }; 	// (4) 	(since C++20)
```

---

Besides that, this function messes up the types of the field. The fields (except `tz`) are now unsigned whereas the parameters of this function are signed.

Reply:
Thanks, haven't heard about aggregate init in c++ :)

Reply:
And now what's the point of having this function, when you can just use the aggregate initialization in place? Or doesn't it work the way you want?

Reply:
This rudiment is removed and cremated. Thanks :)

Reply:
I have eventually figured out the likely reason you still like this function more over the aggregate initialization: the order of fields, yyyy-MM-dd hh:mm:ss in the function vs. ss:hh dd-MM-yyyy in the structure.

I get that. That may be considered a valid reason to have a convenience function like this. Would be nice to explicitly state the benefit (the reason) that function exists, though 🙂

An alternative solution is to rearrange the fields in the structure. It does make some sense to have them arranged from least significant to most significant (though, the actual benefits are unclear to me at the moment), but it also makes sense to arrange them from most significant to least significant components to make the initialization more convenient (a la ISO8601 order).